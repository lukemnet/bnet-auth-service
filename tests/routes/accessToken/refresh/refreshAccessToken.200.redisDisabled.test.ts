const fastify = require('fastify');
const fp = require('fastify-plugin');
const server = require('../../../../src/index');

const config = {
  app: {
    nodeEnv: 'test',
    port: '8123',
  },
  bnet: {
    region: 'us',
    apiKey: 'key',
    apiSecret: 'secret',
  }
}

const redisDisabled = {
  redis: {
    enable: false,
  }
}

describe('/accessToken/refresh (Redis disabled)', () => {
  const fastifyServer = fastify();

  beforeAll(() => {
    fastifyServer.register(server, { ...config, ...redisDisabled });
  });

  beforeEach(() => {
    jest.clearAllMocks();
    jest.resetModules();
  });

  afterEach(() => {
    fastifyServer.close();
  });
 
  it('returns 200', async () => {
    const res = await fastifyServer.inject({ method: 'GET', url: '/accessToken/refresh', });
    expect(res.statusCode).toBe(200);
  });


  it('returns correct response', async () => {
    const res = await fastifyServer.inject({ method: 'GET', url: '/accessToken/refresh', });
    expect(JSON.parse(res.payload)).toEqual({ status:200, message: "Access token not refreshed (Redis disabled)"});
  });
});
