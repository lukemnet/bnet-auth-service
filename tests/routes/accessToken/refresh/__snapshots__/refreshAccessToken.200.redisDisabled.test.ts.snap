// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/accessToken/refresh (Redis disabled) returns correct response 1`] = `"{\\"status\\":200,\\"message\\":\\"Access token not refreshed (Redis disabled)\\"}"`;
